File: LeetCode/1528.Shuffle_String/main.py
#!/usr/bin/python

from solution import Solution

# Unit Test
s1 = "codeleet", indices1 = [4,5,6,7,0,2,1,3]
s2 = "abc", indices2 = [0,1,2]
s3 = "aiohn", indices3 = [3,1,4,2,0]
s = Solution()
if (s.restoreString(s1, indices1) == "leetcode"):
    print("Pass")
else:
    print("Test 1 Failed")
if (s.restoreString(s2, indices2) == "abc"):
    print("Pass")
else:
    print("Test 2 Failed")
if (s.restoreString(s3, indices3) == "nihao"):
    print("Pass")
else:
    print("Test 3 Failed")

File: LeetCode/1528.Shuffle_String/main.cpp
#include "solution.h"

// Unit Test

int main(){
    Solution s;
    vector<int> v1 {4,5,6,7,0,2,1,3};
    vector<int> v2 {0,1,2};
    vector<int> v3 {3,1,4,2,0};
    string s1 = "codeleet";
    string s2 = "abc";
    string s3 = "aiohn";
    if (s.restoreString(s1, v1) == "leetcode") {cout << "Pass" << endl; } else {cout << "Failed 1" << endl; }
    if (s.restoreString(s2, v2) == "abc") {cout << "Pass" << endl; } else {cout << "Failed 2" << endl; }
    if (s.restoreString(s3, v3) == "nihao") {cout << "Pass" << endl; } else {cout << "Failed 3" << endl; }
    
    return 0;
}